app/src/org/commcare/activities/CommCareSetupActivity.java

```diff
@@ -933,4 +941,24 @@ public static String getAnalyticsActionFromInstallMode(int installModeCode) {
                 return "";
         }
     }
+
+    private void checkManagedConfiguration() {
+        Log.d(TAG, "Checking managed configuration");
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
+            // Check for managed configuration
+            RestrictionsManager restrictionsManager =
+                    (RestrictionsManager) getSystemService(Context.RESTRICTIONS_SERVICE);
+            Bundle appRestrictions = restrictionsManager.getApplicationRestrictions();
+            if (appRestrictions.containsKey("profileUrl")) {
+                Toast.makeText(this,  "Found managed configuration install URL "
+                        + appRestrictions.getString("profileUrl"), Toast.LENGTH_LONG).show();
+                Log.d(TAG, "Found managed configuration install URL "
+                        + appRestrictions.getString("profileUrl"));
+                incomingRef = appRestrictions.getString("profileUrl");
+                uiState = UiState.READY_TO_INSTALL;

```

Comment:
Can you create a new install mode value for managed installations (i.e. like `INSTALL_MODE_BARCODE` and `INSTALL_MODE_URL`), and then add a line here that sets the value of `lastInstallMode` to that?